float64 x
float64 y
float64 z

float64 start_x
float64 start_y
float64 start_z
